joomla/joomla-cms
# Improve handling of not supported environments ex. database versions

Pull Request for Issue #38694

### Summary of Changes

As mentiond here #38694 there is no transparent messaging from Joomla when you are running an site which does not meet the minimum requirements for example of the database.

I see here two solutions to this:
- improve the messaging within the CMS
- had bail out of the CMS and show a message similiar to the PHP minimum version

The latter is not b/c and if anything only something for majors to be changed.

### Testing Instructions

- Setup a site that is running MySQL 5.5
- Install Joomla 4.2.0
- Note that an update to 4.2.2 or later will be shown
- But no download URL will be found
- Also check the installer warings (System -> Warnings)
- Confirm that there is no warning showin by the core about that database version issue
- apply this patch
- Check the "We cant find an update" message again it will mention an (to be created once agreed here) doc page instead of a message which does not help to solve the issue.
- Check the warnings tab and confirm there is a clear message showing the issue with the minimum database version.

#### When you dont have a mysql 5.5 or older setup

In that case you can fake the tests mentiond here by doing the following:

Add the following two lines to this file: `administrator\components\com_joomlaupdate\src\View\Joomlaupdate\HtmlView.php` at Line 179-180
```php
        $hasUpdate = true;
        $hasDownload = false;
```

This will make sure you get the "no download URL found all the time".

The seccond check (new warning message) can be faked by reversing the check here after applying the patch: `administrator\components\com_installer\src\Model\WarningsModel.php` at Line 180

from
```php
if (Factory::getDbo()->isMinimumVersion() === false) {
```
to
```php
if (Factory::getDbo()->isMinimumVersion() === true) {
```

### Actual result BEFORE applying this Pull Request

#### We cant find a download URL message
![Screenshot 2022-09-21 140628](https://user-images.githubusercontent.com/2596554/191500479-e6aa7f99-1165-4de2-8af8-a600493c588a.png)

#### Warning within the com_installer warnings

No message at all.

### Expected result AFTER applying this Pull Request

#### We cant find a download URL message
![image](https://user-images.githubusercontent.com/2596554/191498797-6b05c5b8-0c1f-4a62-b7a8-52c31a32a57e.png)

#### Warning within the com_installer warnings
![image](https://user-images.githubusercontent.com/2596554/191498982-a301de2e-8607-4f42-9415-e8ae6db2d8c7.png)

### Documentation Changes Required

The new page dedicated to this issue has to be created maybe under the "Help pages space"?: https://docs.joomla.org/Special:MyLanguage/J4.x:We_cant_find_a_download_url

@coolwebcreations Do you have a preference where to place such new doc page?

Said page should list the possible reasons for this message:
- The update to Joomla %1$s is not available for your stability level (where to find and change that setting)
- the update server setting (index.php?option=com_config&view=component&component=com_joomlaupdate) is not pointing to a official updateserver (where to find and change that setting)
- this environment does not meet the minimum requirements (https://downloads.joomla.org/technical-requirements) for this update (where to find and change that setting)
- there is a problem with the Joomla Update Server. (where to report issues with the update server)

And also mention more deeper ways to check the points mentiond above. Like with the new warning within com_installer as well as that the update can be manually installed. But that should always be the last resort anyway.

By this we keep the message within the CMS as short as possibe but also have a dedicated place to document the issues and solutions.
